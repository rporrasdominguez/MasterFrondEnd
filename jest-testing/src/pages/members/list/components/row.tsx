import * as React from 'react';
import { Member } from '../viewModel';

interface Props {
  member: Member;
}

export const Row: React.StatelessComponent<Props> = (props) => (
  <tr className="item">
    <td>
      <img src={props.member.avatarUrl} style={{ maxWidth: '10rem' }} />
    </td>
    <td>
      <span>{props.member.id}</span>
    </td>
    <td>
      <span>{props.member.name}</span>
    </td>
  </tr>
);
