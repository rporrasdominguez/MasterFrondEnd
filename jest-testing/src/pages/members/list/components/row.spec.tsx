import * as React from 'react';
import {shallow} from 'enzyme';
import { Row } from './row';



describe('pages/members/list/components/body specs', () => {

  it('should render as expected when passing required properties', () => {
    // Arrange
    const props ={
        member:
            {
                id: 1,
                name: 'name test',
                avatarUrl: 'url test',
            }
        
            
    }
    // Act
    const component = shallow(
        <Row {...props}/>
    );
    // Assert
    expect(component).toMatchSnapshot();
  });

});