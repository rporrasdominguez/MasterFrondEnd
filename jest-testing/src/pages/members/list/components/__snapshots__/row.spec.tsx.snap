// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/components/row specs should render as expected when passing required properties 1`] = `
<tr
  className="item"
>
  <td>
    <img
      src="url test"
      style={
        Object {
          "maxWidth": "10rem",
        }
      }
    />
  </td>
  <td>
    <span>
      1
    </span>
  </td>
  <td>
    <span>
      name test
    </span>
  </td>
</tr>
`;
